src/routes/participant/availability/[rehearsalId]/+page.svelte

```diff
@@ -0,0 +1,139 @@
+<script lang="ts">
+	import { db } from '$lib/firebase/client/firebase'
+	import type { availability } from '$lib/types/domain/availability'
+	import type { rehearsal } from '$lib/types/domain/rehearsal'
+	import {
+		Grid,
+		Column,
+		Row,
+		TimePicker,
+		Checkbox,
+		Button,
+		TextInput
+	} from 'carbon-components-svelte'
+	import { Save } from 'carbon-icons-svelte'
+	import { collection, doc, setDoc, Timestamp, updateDoc } from 'firebase/firestore'
+	import { page } from '$app/stores'
+	import { getTimeString } from '$lib/util/timeString'
+
+	export let data: { rehearsal: rehearsal; availability: availability }
+
+	let available: boolean = true
+	let remarksText: string
+	let startTime: string = getTimeString(data.availability.startTime)
+	let endTime: string = getTimeString(data.availability.endTime)
+
+	if (data.availability.reason != null) {
+		remarksText = data.availability.reason
+	}
+
+	function ChangeAvailability() {
+		available = !available
+	}
+
+	async function SaveAvailability() {
+		let a: availability
+
+		if (!available) {
+			a = {
+				available: false,
+				rehearsal: data.availability.rehearsal,
+				reason: remarksText,
+				startTime: Timestamp.fromDate(
+					new Date(
+						data.rehearsal.startTime.toDate().getFullYear(),
+						data.rehearsal.startTime.toDate().getMonth(),
+						data.rehearsal.startTime.toDate().getDate(),
+						+startTime.split(':')[0],
+						+startTime.split(':')[1]
+					)
```

Comment:
Waarom moet dit als je basically de date al hebt in data.rehearsal.startTime.toDate() ?